src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -4108,8 +4108,9 @@ String MergeTreeData::getPartitionIDFromQuery(const ASTPtr & ast, ContextPtr loc
         assert(typeid_cast<ASTFunction *>(partition_ast.value.get()));
         assert(partition_ast.value->as<ASTFunction>()->name == "tuple");
         assert(partition_ast.value->as<ASTFunction>()->arguments);
+        auto args = partition_ast.value->as<ASTFunction>()->arguments;
         bool empty_tuple = partition_ast.value->as<ASTFunction>()->arguments->children.empty();
-        if (!empty_tuple)
+        if (!args || !empty_tuple)
```

Comment:
Probably you wanted to check `args` for nullptr before line 4112, otherwise it does not make sense.